app/routes/companyInterest/components/CompanyInterestList.tsx

```diff
@@ -142,16 +198,15 @@ class CompanyInterestList extends Component<Props, State> {
         };
       }),
     ].sort((o1, o2) => {
-      if (Number(o1.year) === Number(o2.year)) {
-        if (o1.semester === 'spring') {
-          return -1;
-        }
-
-        return 1;
-      }
-
-      return Number(o1.year) > Number(o2.year) ? -1 : 1;
+      return Number(o1.year) === Number(o2.year)
+        ? o1.semester === 'spring'
+          ? -1
+          : 1
+        : Number(o1.year) > Number(o2.year)
+        ? -1
+        : 1;
```

Comment:
This is very difficult to read. The previous way was way easier. Try to iterate on that instead.